src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -860,7 +860,7 @@ public function morphTo($name = null, $type = null, $id = null)
         // If the type value is null it is probably safe to assume we're eager loading
         // the relationship. When that is the case we will pass in a dummy query as
         // there are multiple types in the morph and we can't use single queries.
-        if (is_null($class = $this->$type)) {
+        if (is_null($class = $this->$type) || $class === '') {
```

Comment:
I did that on a previous version which failed tests elsewhere in Model.php.

I will try again after I rebase back to 5.1
On 28 Apr 2016 18:31, "Graham Campbell" <notifications@github.com> wrote:

> In src/Illuminate/Database/Eloquent/Model.php
> <https://github.com/laravel/framework/pull/13360#discussion_r61468802>:
>
> > @@ -860,7 +860,7 @@ public function morphTo($name = null, $type = null, $id = null)
> >          // If the type value is null it is probably safe to assume we're eager loading
> >          // the relationship. When that is the case we will pass in a dummy query as
> >          // there are multiple types in the morph and we can't use single queries.
> > -        if (is_null($class = $this->$type)) {
> > +        if (is_null($class = $this->$type) || $class === '') {
>
> Better to just change this to if (! empty($class = $this->$type)) {
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/13360/files/bc1519926bf4a5a541bfb1aafefe0bcba4f0acbe#r61468802>
>